### PR TITLE
[Bug](materialized-view) enable rewrite on select materialized index with aggregate mode

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
enable rewrite on select materialized index with aggregate mode

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/MaterializedIndexMeta.java [hunk 1]
```diff
@@ -359,4 +359,14 @@ public void initColumnNameMap() {
             definedNameToColumn.put(normalizeName(column.getDefineName()), column);
         }
     }
+
+    // rollup or base column not have define expr
+    public boolean haveDefineExpr() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/MaterializedIndexMeta.java:364]: not use?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/mv/SelectMaterializedIndexWithAggregate.java [hunk 2]
```diff
@@ -689,93 +685,52 @@ private SelectResult select(
...
-            return new SelectResult(PreAggStatus.on(), selectIndexId,
-                    rewriteResultOpt.map(r -> r.exprRewriteMap).orElse(new ExprRewriteMap()));
-        } else {
-            if (scan.getPreAggStatus().isOff()) {
-                return new SelectResult(scan.getPreAggStatus(),
-                        scan.getTable().getBaseIndexId(), new ExprRewriteMap());
-            }
 
-            Set<MaterializedIndex> candidatesWithoutRewriting = new HashSet<>();
+        if (!table.isDupKeysOrMergeOnWrite() && scan.getPreAggStatus().isOff()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/mv/SelectMaterializedIndexWithAggregate.java:689]: could u add some comment to explain why add `scan.getPreAggStatus().isOff()`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/mv/SelectMaterializedIndexWithAggregate.java [hunk 3]
```diff
@@ -689,93 +685,52 @@ private SelectResult select(
...
-                                .collect(Collectors.toList());
-
-                long selectedIndex = selectBestIndex(rollupsWithAllRequiredCols, scan, predicates);
-                if (selectedIndex == scan.getTable().getBaseIndexId()) {
-                    return baseIndexSelectResult;
-                }
-                return new SelectResult(PreAggStatus.on(), selectedIndex, new ExprRewriteMap());
-            }
+        long selectIndexId = selectBestIndex(haveAllRequiredColumns, scan, predicates);
+        if (!table.isDupKeysOrMergeOnWrite() && (new CheckContext(scan, selectIndexId)).isBaseIndex()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/mv/SelectMaterializedIndexWithAggregate.java:724]: could u add some comments to explain what only check `!table.isDupKeysOrMergeOnWrite()`

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run ckbench

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**